### PR TITLE
[explore] include ControlHeader as part of Control interface

### PR DESCRIPTION
**Pull request by mistercrunch**
A bit of code repetition in favor of much more flexibility. Also moving the checkbox to show on one line instead of 2.

### CODE REVIEW

File: superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx [hunk 1]
```diff
@@ -4,29 +4,24 @@ import { Checkbox } from 'react-bootstrap';
...
   beforeEach(() => {
-    wrapper = shallow(<CheckboxControl {...defaultProps} />);
+    wrapper = mount(<CheckboxControl {...defaultProps} />);
   });
 
   it('renders a Checkbox', () => {
     expect(wrapper.find(Checkbox)).to.have.lengthOf(1);
   });
-
-  it('calls onChange when toggled', () => {
```
Comment (mistercrunch) [inline superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx]: Somehow couldn't get simulate to work in the context of a mount (instead of shallow). Gave up after loosing much more time than that test is worth...

File: superset/assets/javascripts/explorev2/stores/controls.jsx [hunk 2]
```diff
@@ -972,13 +972,15 @@ export const controls = {
     type: 'CheckboxControl',
     label: 'Donut',
     default: false,
+    renderTrigger: true,
```
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/stores/controls.jsx]: [Unrelated to PR]

File: superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx (hunk 1 above)
Comment (ascott) [inline superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx]: why the change from shallow to mount?

Comment (mistercrunch) [inline superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx]: Because Checkbox is rendered deeper in a subcomponent

File: superset/assets/javascripts/explorev2/components/ControlHeader.jsx [hunk 3]
```diff
@@ -64,6 +65,7 @@ export default function ControlHeader({
               {' '}
             </span>
           }
+          <span>{leftNode}</span>
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/ControlHeader.jsx]: should this be inside the <ControlLabel /> component? what is the use case for the left node? we probably also want to wrap in a conditional.
```
{leftNode && 
  ...
}
```

File: superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx (hunk 1 above)
Comment (ascott) [inline superset/assets/spec/javascripts/explorev2/components/CheckboxControl_spec.jsx]: gotcha

File: superset/assets/javascripts/explorev2/components/ControlHeader.jsx (hunk 3 above)
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/ControlHeader.jsx]: It's hard to allow for checkbox on a single row without this. The checkbox was way far on the right when using  rightNode....
